### DISCUSSION TITLE
Moving Away from GraphQL (no more sanity-source-plugin)

### DISCUSSION THREAD

**Discussion by iamkevingreen** [ctrl-alt-del-world/midway#20] [Ideas]
I will be moving the core of this experience off of GraphQL, the reasoning here is that the Groq syntax in my eyes is more stable and easier to parse through and handle. It also pairs well with the docs + preview aspects of the Midway build. I find that GraphQL in general is part of the aspects of gatsby that are more cumbersome and don't actually find it helpful. 

I will be adding an Image component that works semi-similar to gatsby-image as that is one of the only benefits in my eyes of utilizing the grapqhl portion of the experience. 

This will also make taking parts of midway into other frameworks like next/react-static a lot easier because it will not have the graphQL syntax handling. 

**NOTE** I have made a fork of the repo that has all the current GraphQL aspects of the build in-tact. If you so choose to it would be pretty easy to always pull from that if you want to use GraphQL and I will also update the readme to reflect that.

**Comment by iamkevingreen** [comment]
This is now in effect, anyone experiencing issues please let me know here, I have not gone through the experience on a fresh install so may be a lil different there but ideally pretty similar experience.

**Reply by HeyMarcy** [reply]
I'm getting an error on `npm install` for the gatsby3 branch:
`npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: midway@0.1.0
npm ERR! Found: groq@2.2.6
npm ERR! node_modules/groq
npm ERR!   groq@"^2.2.6" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer groq@"^0.142.0" from @asbjorn/eslint-plugin-groq@1.0.0
npm ERR! node_modules/@asbjorn/eslint-plugin-groq
npm ERR!   @asbjorn/eslint-plugin-groq@"^1.0.0" from the root project`

**Reply by iamkevingreen** [reply]
the gatsby3 branch is not stable. I don't recommend using that branch until I have more time to iterate/make updates to it.

**Reply by HeyMarcy** [reply]
thanks!
